File: MRNA.cpp
unsigned int mrnaeverse (string rna)
{
  unsigned int count = 3;
  if (rna.length() == 0)
  {
    return count;
  }
  else
  {
    for (int i = 0; i < rna.length(); i++)
    {
      switch(rna[i])
      {
        case 'F':
          count = count * 2;
          break;
        case 'L':
          count = count * 6;
          break;
        case 'S':
          count = count * 6;
          break;
        case 'P':
          count = count * 4;
          break;
        case 'Y':
          count = count * 2;
          break;
        case 'I':
          count = count * 3;
          break;
        case 'V':
          count = count * 4;
          break;
        case 'A':
          count = count * 4;
          break;
        case 'T':
          count = count * 4;
          break;
        case 'G':
          count = count * 4;
          break;
        case 'R':
          count = count * 6;
          break;
        case 'K':
          count = count * 2;
          break;
        case 'D':
          count = count * 2;
          break;
        case 'N':
          count = count * 2;
          break;
        case 'E':
          count = count * 2;
          break;
        case 'Q':
          count = count * 2;
          break;
        case 'H':
          count = count * 2;
          break;
        case 'C':
          count = count * 2;
          break;
        case 'W':
          count = count * 1;
          break;
        case 'M':
          count = count * 1;
          break;
      }
    }
  }
  return count;
}

void main()
{
  string mrna;
  cout << "enter the mRNA" << endl;
  cin >> mrna;
  unsigned int number = mrnaeverse (mrna);
  cout << number << endl;
}

File: MRNA.cs
using System;
  
public class Program
{
  public static int Main()
  {
    string mrna;
    Console.WriteLine("Enter the mRNA")
    mrna = Console.ReadLine();
    long number = mrnareverse (mrna);
    Console.WriteLine(number);
  }
  public static long mrnareverse(string rna)
  {
    long count = 3;
    List<Tuple<char, int>>map = new List<Tuple<char, int>>();
    map.Add(new Tuple<char, int>('F', 2));
    map.Add(new Tuple<char, int>('L', 6));
    map.Add(new Tuple<char, int>('S', 6));
    map.Add(new Tuple<char, int>('P', 4));
    map.Add(new Tuple<char, int>('Y', 2));
    map.Add(new Tuple<char, int>('I', 3));
    map.Add(new Tuple<char, int>('V', 4));
    map.Add(new Tuple<char, int>('A', 4));
    map.Add(new Tuple<char, int>('T', 4));
    map.Add(new Tuple<char, int>('G', 4));
    map.Add(new Tuple<char, int>('R', 6));
    map.Add(new Tuple<char, int>('K', 2)); 
    map.Add(new Tuple<char, int>('D', 2));
    map.Add(new Tuple<char, int>('N', 2));
    map.Add(new Tuple<char, int>('E', 2));
    map.Add(new Tuple<char, int>('Q', 2));
    map.Add(new Tuple<char, int>('H', 2));
    map.Add(new Tuple<char, int>('C', 2));
    map.Add(new Tuple<char, int>('W', 1));
    map.Add(new Tuple<char, int>('M', 1));
    for (int i = 0; i<rna.Length; i++)
    {
      Tuple<char, int> x = map.Find(x=>x.Item1 == rna[i]);
      count = count *x.Item2;
    }
    return count;
  }
}
